### DISCUSSION TITLE
Bluetooth for ESP32 with MQTT TLS

### DISCUSSION THREAD

**Discussion by github768484** [arendst/Tasmota#18044] [Q&A]
I cannot get my complied build to load onto the ESP32. I am using gitpod and edited the following files. The tasmota.bin file seems small @699KB. Ultimately, I want to have a BLE enabled esp32 capable of loading TlsKey1 and TlsKey2 for connection to my private MQTT bridge.

/workspace/Tasmota/tasmota/user_config_override.h
#ifndef USE_MQTT_TLS
#define USE_MQTT_TLS
#define USE_MQTT_TLS_CA_CERT // Optional but highly recommended
#endif 

/workspace/Tasmota/platformio_override.ini
...
;                          tasmota
                          tasmota32-bluetooth
;                          tasmota32-webcam
...

**Comment by sfromis** [comment]
You should not look for `tasmota.bin` as this is ESP8266, while the file for ESP32 should be `tasmota32.bin`

**Reply by github768484** [reply]
I'm not sure where I might find tasmota32.bin when compiling with gitpod. Here is the path as I know it. 

/workspace/Tasmota/build_output/firmware/tasmota.bin

This very well may be an identification of error, but I believe that my modification of platformio_override.ini is pulling the correct build.

**Reply by sfromis** [reply]
If you do not find `tasmota32.bin` in that folder, your Gitpod compile was likely selecting the build environment `tasmota` instead of `tasmota32`. This selection is not bound to be what default you may have in `platformio_override.ini`. Of course, you can check the time stamp of the build you found to verify that it is indeed the most recent build, not a file lingering from previously.

**Comment by github768484** [comment]
First of all, Thank you for your reply. 

There is another bin that I found from the build output. However, it is the same size at the tasmota.bin. Gitpod is not providing timeline information on these. Any other ideas?

`Building .pio/build/tasmota/firmware.bin`

Additionally, the built seems to be using the overrrides. 

```
Building in release mode
*** use provided user_config_override.h as planned ***
*** use provided platformio_override.ini as planned ***
*** use provided platformio_tasmota_cenv.ini as planned ***
```

**Reply by barbudor** [reply]
1) to compile tasmota32 you can either select it in the environment list somewhere in the status line approx 1/3rd to 1/2 from the left of the window
Or by typing command: `platformio run -e tasmota32`

2) you are doing that for nothing as TLS is included in every tasmota32 variant including tasmota32-bluetooth
Beside, USE_MQTT_TLS_CA_CERT is obsolete and has been replaced by setoption132

**Reply by github768484** [reply]
It appears that I'm in the weeds here. All I want to do is to connect a tasmota32bluetooth to my AWS MQTT. I have confirmation of a working NodeRed client. And according to this tutorial [Self signed Mosquitto](https://tasmota.github.io/docs/Self-signed-Mosquitto/) I need to load TLSKey1 and TLSKey2 onto Tasmota via the console. However, I get an unknown command. I thought that I needed to enable TLS via a custom binary.

**Reply by sfromis** [reply]
I was not suggesting timeline info *during* build, but checking file time stamp when observing the file *after* the build completed. While this is "surprisingly unintuitive", if you right-click on the folder/file and select "Open in Integrated Terminal", you can then execute the command `ls -l` to get a directory listing including time stamp. Not sure if there's a better way, but this worked for me.

My expectation would be that you'd not be seeing it to be some old file, but that you **really** built the ESP8266 version instead of the ESP32 version you wanted. Hence me using the word "verify".

Anyway, better not look at `firmware.bin` which is a "temporary" artifact during build, you were looking correctly in `build_output`.

**Reply by sfromis** [reply]
To enable TLS in a precompiled ESP32 Tasmota build, use `SetOption103`.

**Reply by barbudor** [reply]
And [Setoption132 0](https://tasmota.github.io/docs/Commands/#setoption132) to enable CA Cert mode as required for AWS IOT
